java/arcs/core/allocator/Arc.kt

```diff
@@ -173,9 +173,18 @@ class Arc internal constructor(
     suspend fun waitForStart() = waitFor(Running)
 
     /** Stop the current [Arc]. */
-    suspend fun stop() = allocator.stopArc(id)
+    suspend fun stop() = allocator.stopArc(id).also {
+        onArcStateChangeFiltered(Stopped) { closeFlow() }
+    }
 
-    @Suppress("UNUSED_PARAMETER") private fun <T> sync(obj: Any, block: () -> T) = block()
+    private fun unregisterChangeHandlerWithArcHosts() {
+        runBlocking {
```

Comment:
I remember `runBlocking` (supports only JVM and native) cannot be used at `arcs.core.*` for cross-platform transpilation. Not sure whether it is still true or not.

Reply:
it's still true, KotlinJS can't support it. I'll launch a coroutine.